File: linkis-commons/linkis-storage/src/main/java/org/apache/linkis/storage/domain/MethodEntity.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.linkis.storage.domain;

/**
 * Engine unique Id(engine唯一的Id)
 *
 * <p>Fs type(fs类型)
 *
 * <p>Create a user to start the corresponding jvm user(创建用户为对应启动的jvm用户)
 *
 * <p>Proxy user(代理用户)
 *
 * <p>client Ip for whitelist control(ip用于白名单控制)
 *
 * <p>Method name called(调用的方法名)
 *
 * <p>Method parameter(方法参数)
 */
public class MethodEntity {
  private long id;
  private String fsType;
  private String creatorUser;
  private String proxyUser;
  private String clientIp;
  private String methodName;
  private Object[] params;

  public MethodEntity(
      long id,
      String fsType,
      String creatorUser,
      String proxyUser,
      String clientIp,
      String methodName,
      Object[] params) {
    this.id = id;
    this.fsType = fsType;
    this.creatorUser = creatorUser;
    this.proxyUser = proxyUser;
    this.clientIp = clientIp;
    this.methodName = methodName;
    this.params = params;
  }

  public long getId() {
    return id;
  }

  public void setId(long id) {
    this.id = id;
  }

  public String getFsType() {
    return fsType;
  }

  public void setFsType(String fsType) {
    this.fsType = fsType;
  }

  public String getCreatorUser() {
    return creatorUser;
  }

  public void setCreatorUser(String creatorUser) {
    this.creatorUser = creatorUser;
  }

  public String getProxyUser() {
    return proxyUser;
  }

  public void setProxyUser(String proxyUser) {
    this.proxyUser = proxyUser;
  }

  public String getClientIp() {
    return clientIp;
  }

  public void setClientIp(String clientIp) {
    this.clientIp = clientIp;
  }

  public String getMethodName() {
    return methodName;
  }

  public void setMethodName(String methodName) {
    this.methodName = methodName;
  }

  public Object[] getParams() {
    return params;
  }

  public void setParams(Object[] params) {
    this.params = params;
  }

  @Override
  public String toString() {
    return "id:"
        + id
        + ", methodName:"
        + methodName
        + ", fsType:"
        + fsType
        + ", creatorUser:"
        + creatorUser
        + ", proxyUser:"
        + proxyUser
        + ", clientIp:"
        + clientIp;
  }
}

File: linkis-commons/linkis-storage/src/main/scala/org/apache/linkis/storage/domain/MethodEntity.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.linkis.storage.domain

import java.lang.reflect.Type

import com.google.gson.{GsonBuilder, ToNumberPolicy}

/**
 * @param id
 *   Engine unique Id(engine唯一的Id)
 * @param fsType
 *   Fs type(fs类型)
 * @param creatorUser
 *   Create a user to start the corresponding jvm user(创建用户为对应启动的jvm用户)
 * @param proxyUser
 *   Proxy user(代理用户)
 * @param clientIp
 *   client Ip for whitelist control(ip用于白名单控制)
 * @param methodName
 *   Method name called(调用的方法名)
 * @param params
 *   Method parameter(方法参数)
 */
case class MethodEntity(
    id: Long,
    fsType: String,
    creatorUser: String,
    proxyUser: String,
    clientIp: String,
    methodName: String,
    params: Array[AnyRef]
) {

  override def toString: String = {
    s"id:$id, methodName:$methodName, fsType:$fsType, " +
      s"creatorUser:$creatorUser, proxyUser:$proxyUser, clientIp:$clientIp, "
  }

}

object MethodEntitySerializer {

  val gson = new GsonBuilder()
    .setDateFormat("yyyy-MM-dd'T'HH:mm:ssZ")
    .setObjectToNumberStrategy(ToNumberPolicy.LAZILY_PARSED_NUMBER)
    .create

  /**
   * Serialized to code as a MethodEntity object 序列化为code为MethodEntity对象
   * @param code
   * @return
   */
  def deserializer(code: String): MethodEntity = gson.fromJson(code, classOf[MethodEntity])

  /**
   * Serialize MethodEntity to code 序列化MethodEntity为code
   * @param methodEntity
   * @return
   */
  def serializer(methodEntity: MethodEntity): String = gson.toJson(methodEntity)

  /**
   * Serialize a java object as a string 序列化java对象为字符串
   * @param value
   * @return
   */
  def serializerJavaObject(value: Any): String = gson.toJson(value)

  /**
   * Deserialize a string into a java object 将字符串解序列化为java对象
   * @param json
   * @param classType
   * @tparam T
   * @return
   */
  def deserializerToJavaObject[T](json: String, classType: Class[T]): T = {
    gson.fromJson(json, classType)
  }

  def deserializerToJavaObject[T](json: String, oType: Type): T = {
    gson.fromJson(json, oType)
  }

}
